lightly/api/api_workflow_upload_dataset.py

```diff
@@ -54,18 +49,13 @@ def upload_dataset(self, input: Union[str, LightlyDataset], max_workers: int = 8
                              f"but is of type {type(input)}")
 
         # check the allowed dataset size
-        api_max_dataset_size, status_code = get_quota(self.token)
-        max_dataset_size = min(api_max_dataset_size, LIGHTLY_MAXIMUM_DATASET_SIZE)
+        max_dataset_size_str = self.quota_api.get_quota_maximum_dataset_size()
```

Comment:
Just a question: If the call to the api fails, will the method throw an error or will it simply return the message from the api? In the second case, I'd add some error handling here, otherwise you get an ugly error message like: "Can't convert `max_dataset_size_str` to int"

Reply:
If the URL is nonexistent, the following error occurs. Thus I would keep it like that.
```
2021-03-15 09:50:44,408 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c2d1640>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,411 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c2d1970>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,413 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c2d1ac0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,420 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30d070>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,420 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30d070>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,422 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c2d1a60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,422 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c2d1a60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,424 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c2d19a0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,424 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c2d19a0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/latest?currentVersion=1.1.1
2021-03-15 09:50:44,434 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30da60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
2021-03-15 09:50:44,434 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30da60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
2021-03-15 09:50:44,434 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30da60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
2021-03-15 09:50:44,436 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30db20>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
2021-03-15 09:50:44,436 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30db20>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
2021-03-15 09:50:44,436 WARNING Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30db20>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
2021-03-15 09:50:44,439 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30dc70>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
2021-03-15 09:50:44,439 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30dc70>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
2021-03-15 09:50:44,439 WARNING Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x14c30dc70>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /v1/versions/pip/minimum
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socket.py", line 953, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```

Reply:
I don't mean if the URL is nonexistent but if the endpoint returns a 500 or whatever because of a server side problem.

Reply:
The generated code either returns the payload of the 200-response or it raises an error.
E.g. if the token is wrong, the following error is raised:
```
lightly.openapi_generated.swagger_client.rest.ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'Access-Control-Allow-Origin': '*', 'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '60', 'ETag': 'W/"3c-2MASddOo7f/Y1UTHEz2BMijz4sE"', 'X-Response-Time': '14.234ms', 'Date': 'Mon, 15 Mar 2021 09:43:49 GMT', 'Connection': 'keep-alive', 'Keep-Alive': 'timeout=5'})
HTTP response body: {
    "error": "Not able to authenticate user using token"
}
```